pkg/operator/k8sutil/k8sutil.go

```diff
@@ -63,3 +63,11 @@ func GetK8SVersion(clientset kubernetes.Interface) (*version.Version, error) {
 	}
 	return version.MustParseSemantic(serverVersion.GitVersion), nil
 }
+
+// TruncateName truncates a string to 63 characters as is required for most names in Kubernetes
+func TruncateName(name string) string {
+	if len(name) > 63 {
```

Comment:
a pod name will be generated from the batch job, which will make the pod name even longer. Will the pod name be sufficiently truncated with this limit on the batch job name?

Reply:
Not a problem:
Job name is `rook-ceph-osd-prepare-k8s-worker-23.this.is.a.very.long.example` (63 long):
```
rook-ceph-osd-prepare-k8s-worker-23.this.is.a.very.long.ex9s58m   0/1       Completed   0          17s
```

Reply:
As discussed, there are a few considerations here when truncating the name:
- If names are truncated we run the risk of multiple nodes having the same resource name, which won't work. If we take a SHA1 hash of the node name and include it in place of the node name, it will be unique.
- To avoid issues during upgrade, we would only want to take a hash of the node name when it is too long.
- When the hash is included in the resource name, we need to find a way to associate the resource with the original node name. If labels won't work because of the length, we need to find another way for troubleshooting.